src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IRequestContextFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface IRequestContextFactory<TRequestContext>
+{
+    Task<TRequestContext?> CreateRequestContextAsync(IQueueItem<TRequestContext> queueItem, CancellationToken queueCancellationToken, CancellationToken requestCancellationToken);
```

Comment:
these tyeps and methods needs docs, even with potential examples of lsp requests to help make this clearer.